search-solr/src/test/java/edu/unc/lib/boxc/search/solr/services/SearchStateFactoryTest.java

```diff
@@ -265,4 +277,111 @@ public void extractAdvancedSearchCollectionIdRight() {
         assertEquals("40000", searchState.getSearchFields()
                 .get(SearchFieldKey.COLLECTION_ID.name()));
     }
+
+    @Test
+    public void extractFacetLimitsValid() {
+        Map<String, String[]> parameters = new LinkedHashMap<>();
+        parameters.put("anywhere", new String[]{""});
+        parameters.put("facetLimits", new String[]{"language:20"});
+
+        SearchState searchState = searchStateFactory.createSearchState(parameters);
+        var limits = searchState.getFacetLimits();
+        assertEquals(1, limits.size());
+        assertEquals(20, limits.get("LANGUAGE"));
+    }
+
+    @Test
+    public void extractFacetLimitsUnknownField() {
+        Map<String, String[]> parameters = new LinkedHashMap<>();
+        parameters.put("anywhere", new String[]{""});
+        parameters.put("facetLimits", new String[]{"boxyfield:20"});
+
+        SearchState searchState = searchStateFactory.createSearchState(parameters);
+        assertTrue(searchState.getFacetLimits().isEmpty());
+    }
+
+    @Test
+    public void extractFacetLimitsInvalidFacetClass() {
+        Map<String, String[]> parameters = new LinkedHashMap<>();
+        parameters.put("anywhere", new String[]{""});
+        parameters.put("facetLimits", new String[]{"language:20"});
+
+        try (MockedStatic<SearchSettings> mockedStatic = mockStatic(SearchSettings.class)) {
```

Comment:
what does MockedStatic do?

Reply:
It lets you mock a static method, which is happening on the next line down. Since SearchSettings.getFacetClass is not a instance method, we can't use a regular mocked object to change its return type.